connectors/agent/service_manager.py

```diff
@@ -0,0 +1,91 @@
+from connectors.logger import DocumentLogger
+from connectors.services.base import (
+    ServiceAlreadyRunningError,
+    get_services,
+)
+from connectors.utils import CancellableSleeps
+
+
+class ConnectorServiceManager:
+    """Class responsible for properly configuring and running Connectors Service in Elastic Agent
+
+    ConnectorServiceManager is a middle man between Elastic Agent and ConnectorsService.
+
+    This class is taking care of starting Connectors Service subservices with correct configuration.
+    If configuration changes, as reported by Agent, then this Manager class gracefully shuts down the
+    subservices and starts them again with new configuration.
+
+    """
+
+    name = "connector-service-manager"
+
+    def __init__(self, configuration):
+        """Inits ConnectorServiceManager with shared ConnectorsAgentConfigurationWrapper.
+
+        This service is supposed to be ran once, and after it's stopped or finished running it's not
+        supposed to be started again.
+
+        There is nothing enforcing it, but expect problems if that happens.
+        """
+        service_name = self.__class__.name
+        self._logger = DocumentLogger(
+            f"[{service_name}]", {"service_name": service_name}
+        )
+        self._agent_config = configuration
+        self._multi_service = None
+        self._running = False
+        self._sleeps = CancellableSleeps()
+
+    async def run(self):
+        """Starts the running loop of the service.
+
+        Once started, the service attempts to run all needed connector subservices
+        in parallel via MultiService.
+
+        Service can be restarted - it will keep running but with refreshed config,
+        or it can be stopped - it will just gracefully shut down.
+        """
+        if self._running:
+            msg = f"{self.__class__.__name__} is already running."
+            raise ServiceAlreadyRunningError(msg)
```

Comment:
should we add the error log, I'm not sure we catch-and-log the exception anywhere

Reply:
This one is not catched, it'll just crash the caller. Supposedly this case is "never gonna happen", so I'd just keep it like this.